dateparser/languages/locale.py

```diff
@@ -109,6 +109,31 @@ def clean_dictionary(dictionary, threshold=2):
             del dictionary[del_key]
         return dictionary
 
+    @property
+    def weekdays(self):
+        weekdays = [
+            "monday",
+            "tuesday",
+            "wednesday",
+            "thursday",
+            "friday",
+            "saturday",
+            "sunday",
+        ]
+        return weekdays
+
+    def remove_multiple_occurrences(self, date_str_tokens: list):
+        # first occurrence of day of the week will be considered
+        # followings occurrence(s) will be skipped and removed from the token list.
+        weekdays_counter = 0
+        for i, token in enumerate(date_str_tokens):
+            if token in self.weekdays:
+                weekdays_counter += 1
+
+                if weekdays_counter > 1:
+                    date_str_tokens.pop(i)
+                    continue
```

Comment:
Interesting solution. I imagine it is not perfect, but if current tests pass, onward! We can always iterate on it later.